File: server/src/main/java/com/xcore/server/configs/WebSecurityConfig.java
package com.xcore.server.configs;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;

/*
 * TODO: Proper security for application.
 *
 * Docs:
 * https://docs.spring.io/spring-security/site/docs/current/reference/html/jc.html
 */

@Configuration
@EnableWebSecurity
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {

  @Autowired
  ApplicationConfig applicationConfig;

  @Override
  protected void configure(HttpSecurity httpSecurity) throws Exception {

    // Disable csrf because we will use token for our api requests.
    httpSecurity.csrf().disable();

    httpSecurity.authorizeRequests().antMatchers("/**").permitAll();
  }

}

File: modules/x_core_server_application/src/main/scala/com/xcore/application/modules/authentication/configs/WebSecurityConfig.scala
package com.xcore.application.modules.authentication.configs

import com.xcore.application.modules.authentication.services.ApplicationUserDetailService
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.crypto.password.PasswordEncoder

@Configuration
@EnableWebSecurity(debug = false)
@EnableGlobalMethodSecurity(prePostEnabled = true)
class WebSecurityConfig extends WebSecurityConfigurerAdapter {

  private val log: Logger = LoggerFactory.getLogger("[🔒SECURITY]");

  @Autowired
  private var appUserDetailsService: ApplicationUserDetailService = _;

  @Autowired
  private var passwordEncoder: PasswordEncoder = _;

  /*
   * Configuration:
   */

  @Bean
  override def authenticationManager: AuthenticationManager = super.authenticationManager;

  @throws[Exception]
  override def configure(authenticationManagerBuilder: AuthenticationManagerBuilder): Unit = {

    log.info("Configuring application security manager.");

    authenticationManagerBuilder
      .userDetailsService(appUserDetailsService)
      .passwordEncoder(passwordEncoder);
  }

  @throws[Exception]
  override def configure(httpSecurity: HttpSecurity): Unit = {

    log.info("Configuring httpSecurity and session management.")

    httpSecurity
      .cors().disable()
      .csrf().disable();
    }

}
